rust/otap-dataflow/crates/engine/README.md

```diff
@@ -2,46 +2,70 @@
 
 Status: **WIP**
 
-## Nodes
-
-The Pipeline Engine consists of three main node types:
-
-- Receiver: A node that receives or scrapes telemetry data from a telemetry
-  source.
-- Processor: A node that processes telemetry data.
-- Exporter: A node that exports telemetry data to a telemetry sink.
-
-Each node can receive control messages from the Pipeline Engine to manage its
-operation.
-
-```mermaid
-graph LR
-    RecvControlIn[Control Channel] --> Receiver
-    Receiver((Receiver))
-    Receiver --> RecvControlOut[Control Channel]
-    Receiver --> RecvDataOut[Data Channel]
-    class Receiver node
-    class RecvControlIn,RecvControlOut controlQueue
-    class RecvDataOut dataQueue
-    
-    ProcControlIn[Control Channel] --> Processor
-    ProcDataIn[Data Channel] --> Processor
-    Processor((Processor))
-    Processor --> ProcControlOut[Control Channel]
-    Processor --> ProcDataOut[Data Channel]
-    class Processor node
-    class ProcControlIn,ProcControlOut controlQueue
-    class ProcDataIn,ProcDataOut dataQueue
-
-    ExControlIn[Control Channel] --> Exporter
-    ExDataIn[Data Channel] --> Exporter
-    Exporter((Exporter))
-    Exporter --> ExControlOut[Control Channel]
-    class Exporter node
-    class ExControlIn,ExControlOut controlQueue
-    class ExDataIn,ExDataOut dataQueue
-
-    classDef controlQueue fill:#ffdddd,stroke:#cc0000,color:#000,stroke-width:0px,font-size:10px,padding:0px
-    classDef dataQueue fill:#ddffdd,stroke:#009900,color:#000,stroke-width:0px,font-size:10px,padding:0px
-    classDef node fill:#4a90e2,color:#ffffff,stroke-width:0px,font-size:12px
-```
+## Introduction
+
+The term pipeline is used here to represent an interconnection of nodes forming
+a Directed Acyclic Graph (DAG). The inputs of a pipeline are called receivers,
+the intermediate processing nodes are called processors, and the output nodes
+are referred to as exporters.
+
+Messages flowing through a pipeline are generically referred to as pdata (i.e.
+Pipeline Data). An OTLP message or an OTAP message are examples of pdata types.
+This pipeline framework is generic over pdata, which means:
+
+- It is possible to instantiate an OTLP pipeline, an OTAP pipeline, or even a
+  pipeline designed to support another type of pdata.
+- It is not possible to support multiple pdata types within a single pipeline.
+  However, a fourth type of component, called a connector, can be used to bridge
+  two pipelines and enable interoperability between different pdata types.
+
+This terminology aligns well with the concepts introduced in the OTEL Go
+Collector.
+
+## Architecture
+
+A list of the design principles followed by this project can be found
+[here](../../docs/design-principles.md). More specifically, the pipeline engine
+implemented in this crate follows a share-nothing, thread-per-core approach.
+In particular, one instance of the pipeline engine is created per core. This
+engine:
+
+- Is based on a single-threaded async runtime
```

Comment:
Just for my understanding, this means that we create a single threaded runtime for each core, right?

Reply:
Yes